### PR TITLE
Adds support for monitoring UDP and TCP DNS lookups

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/7813

To better monitor issues with TCP DNS lookups, we now explicitly make UDP and TCP based DNS lookups.

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
It makes more sense to have the protocol as a label on the histogram, but we don't currently support this in histogramvec.

I wanna add support to histogramvec for multiple labels, but that's a more complicated change. So, I wanna go with this, and refactor in future once we have that larger change done.

See https://github.com/giantswarm/giantswarm/issues/7822 for tracking.

File: dns/dns.go [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 import (
-	"net"
+	"fmt"
 	"sync"
 	"time"
 
 	"github.com/giantswarm/exporterkit/histogramvec"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
+	dnsclient "github.com/miekg/dns"
```
Comment (kopiczko) [inline dns/dns.go]: Do we have that in `Gopkg.toml`?

File: dns/dns.go [hunk 2]
```diff
@@ -21,41 +24,68 @@ const (
 
 // Config provides the necessary configuration for creating a Collector.
 type Config struct {
-	Logger micrologger.Logger
+	KubernetesClient kubernetes.Interface
```
Comment (kopiczko) [inline dns/dns.go]:
```suggestion
	K8sClient kubernetes.Interface
```

File: dns/dns.go (hunk 1 above)
Comment (JosephSalisbury) [reply dns/dns.go]: We do now!

File: dns/dns.go (hunk 2 above)
Comment (JosephSalisbury) [reply dns/dns.go]: Will take care in follow up, this is also differently named in the network collector, see https://github.com/giantswarm/giantswarm/issues/7823 for tracking

**Pull request merged**